system/jlib/jevent.cpp

```diff
@@ -1757,6 +1821,513 @@ IEventVisitor* createDumpCSVEventVisitor(IBufferedSerialOutputStream& out)
+            {
+                if (tmp.compare(pattern) == 0)
+                    return true;
+                if (isWildString(pattern.c_str()))
+                {
+                    if (WildMatch(value, pattern.c_str(), false))
+                        return true;
+                }
+                else
+                {
+                    try
+                    {
+                        RegExpr re(pattern.c_str(), false);
+                        if (re.find(value))
+                            return true;
+                    }
+                    catch (IException* e)
+                    {
+                        // Ignore exception and try next pattern
+                        e->Release();
+                    }
+                }
+            }
+            return false;
+        }
+
+        bool acceptToken(const char* token)
+        {
+            patterns.insert(token);
+            return true;
+        }
+    };
+
+    struct FilterTerm : public CInterface
+    {
+        virtual bool matches(EventType evt, const char* value) const { return true; }
+        virtual bool matches(EventType evt, bool value) const { return true; }
+        virtual bool matches(EventType evt, __uint64 value) const { return true; }
+        virtual bool accept(const char* values)
+        {
+            StringArray tokens;
+            tokens.appendList(values, ",", true);
+            ForEachItemIn(i, tokens)
+            {
+                if (!acceptToken(tokens.item(i)))
+                    return false;
+            }
+            return true;
+        }
+        virtual bool acceptToken(const char* token) = 0;
+    };
+
+    struct StringFilterTerm : public FilterTerm
+    {
+        StringMatchHelper helper;
+        bool matches(EventType evt, const char* value) const override
+        {
+            return helper.matches(value);
+        }
+
+        bool acceptToken(const char* token) override
+        {
+            return helper.acceptToken(token);
+        }
+    };
+
+    struct BoolFilterTerm : public FilterTerm
+    {
+        bool acceptTrue{false};
+        bool acceptFalse{false};
+
+        bool matches(EventType evt, bool value) const override
+        {
+            return (value && acceptTrue) || (!value && acceptFalse);
+        }
+
+        bool acceptToken(const char* token) override
+        {
+            bool value = strToBool(token);
+            if (value && !acceptTrue)
+                acceptTrue = true;
+            if (!value && !acceptFalse)
+                acceptFalse = true;
+            return true;
+        }
+    };
+
+    struct UnsignedFilterTerm : public FilterTerm
+    {
+        using Accepted = std::set<std::pair<__uint64, __uint64>>;
+        Accepted accepted;
+
+        bool matches(EventType evt, __uint64 value) const override
+        {
+            for (const std::pair<__uint64, __uint64>& range : accepted)
+            {
+                if (value > range.second)
```

Comment:
Should have a comment why this is here.  The code would read more easily if this followed the range test.
I am not sure if is worth including, and/or whether it is worth also having a unordered set of exact matches..  Leave in for now.

Reply:
Will remove. Adding a comment made me think about an assumption that has to be made for this to be valid. There is no logic in acceptToken to prevent overlapping ranges. Exceeding the upper bound of one range does not mean there isn't a subsequent range that will match.

In normal use, multiple ranges will probably be rare. Enough ranges to make the attempted optimization significant is unlikely to ever happen.